### PR TITLE
Improving formatting support and outputting well-formed HTML.

### PR DESCRIPTION
**Pull request by tjaderxyz**
Added support for color, hex color, reverse color, strikethrough and monospace.

Used https://modern.ircdocs.horse/formatting.html as reference.

I made it always output well-formed HTML by closing all tags whenever there's a control character and reopening them again. This leads to some unnecessarily verbose output, for example (I'm using [B] for \x02, [I] for \x1d, etc):

`[B][I]bold italic` would lead to `<b></b><b><i>bold italic</i></b>`
`[C]02[I]color italic` would lead to `<font mx-data-color='#00007f'></font><font mx-data-color='#00007f'><i>bold italic</i></font>`

In my opinion these unnecessary empty tags are better than the current situation with malformed HTML. I could get rid of them by keeping track of what has been opened in a stack and only closing tags whenever necessary instead of always, but that would make the code a bit more complex. If you think it would be better I can implement it.

### CODE REVIEW

File: heisenbridge/network_room.py [hunk 1]
```diff
@@ -132,6 +133,7 @@ def init(self):
         self.pills_length = 2
         self.pills_ignore = []
         self.autoquery = True
+        self.color = True
```
Comment (hifi) [inline heisenbridge/network_room.py:136]: I wonder if we should plan to have Matrix->IRC color support as well and rename this to something like `irc_colors` _or_ store it like an enum `[None, "irc", "matrix", "both"]` (this is a pattern that's already used) as it becomes ambiguous when expanded.

File: heisenbridge/private_room.py [hunk 2]
```diff
@@ -46,63 +47,198 @@ def wrapper(*args, **kwargs):
...
+    background = None
+    reversed = False
+    monospace = False
     italic = False
+    strikethrough = False
     underline = False
 
     for m in re.finditer(
-        r"(\x02|\x03([0-9]{1,2})?(,([0-9]{1,2}))?|\x1D|\x1F|\x16|\x0F)?([^\x02\x03\x1D\x1F\x16\x0F]*)", input
+        r"(\x02|\x03(?:([0-9]{1,2})(?:,([0-9]{1,2}))?)?|\x04(?:([0-9A-Fa-f]{6})(?:,([0-9A-Fa-f]{6}))?)?|\x1D|\x1E|\x1F|\x16|\x0F)?([^\x02\x03\x04\x1D\x1E\x1F\x16\x0F]*)", input
```
Comment (hifi) [inline heisenbridge/private_room.py:177]: :cold_sweat: :+1:

**Review by hifi — COMMENTED** [review]
Empty tags are definitely better than broken html, thanks for fixing it!

I'll also need to run this to see the results but looking fairly good so far.

File: heisenbridge/network_room.py (hunk 1 above)
Comment (tjaderxyz) [reply heisenbridge/network_room.py:136]: I don't plan to implement Matrix->IRC color support, so I leave the decision of how to store the variable to you.

File: heisenbridge/private_room.py (hunk 2 above)
Comment (tjaderxyz) [reply heisenbridge/private_room.py:177]: :cold_sweat: indeed, I even forgot to include the code for monospace.

**Comment by hifi** [comment]
Tested and it works good, thanks!

**Pull request merged**